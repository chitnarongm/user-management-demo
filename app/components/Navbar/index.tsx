"use client";
import LogOutIcon from "@mui/icons-material/Logout";
import { AppBar, Box, Button, Toolbar, Typography } from "@mui/material";
import { signOut, useSession } from "next-auth/react";
import { FunctionComponent } from "react";

const Navbar: FunctionComponent = () => {
  const { data: session } = useSession();

  const handleLogOut = () => {
    signOut({ callbackUrl: "/login" });
  };

  return (
    <Box data-testid="navbar" sx={{ maxHeight: "64px", width: "100%", position: "fixed", top: 0, zIndex: "1100" }}>
      <AppBar position="static">
        <Toolbar>
          <Typography data-testid="navbar-title" variant="h5" component="div" sx={{ flexGrow: 1 }}>
            User Management
          </Typography>
          {session && (
            <Box display="flex" alignItems="center">
              <Box marginRight="24px">
                <Typography data-testid="user-info" lineHeight={0.75} fontWeight="bold" variant="subtitle2">
                  {session.user?.email}
                </Typography>
              </Box>
              <Button data-testid="logout-button" onClick={handleLogOut} color="inherit" startIcon={<LogOutIcon />}>
                Log Out
              </Button>
            </Box>
          )}
        </Toolbar>
      </AppBar>
    </Box>
  );
};

export default Navbar;
