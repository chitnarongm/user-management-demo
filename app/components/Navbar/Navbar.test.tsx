import { render, screen } from "@testing-library/react";
import * as nextAuth from "next-auth/react";
import Navbar from ".";

describe("Navbar", () => {
  beforeEach(() => {
    jest
      .spyOn(nextAuth, "useSession")
      .mockReturnValue({ data: { user: { email: "mockuser@email.com" } }, status: "authenticated" } as any);
  });
  const renderNavbar = () => {
    return render(<Navbar />);
  };

  it("should render correctly", () => {
    const { asFragment } = renderNavbar();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display information correctly when user session is existed", () => {
    renderNavbar();

    expect(screen.getByTestId("navbar")).toBeVisible();
    expect(screen.getByTestId("navbar-title")).toHaveTextContent("User Management");
    expect(screen.getByTestId("user-info")).toHaveTextContent("mockuser@email.com");
    expect(screen.getByTestId("logout-button")).toBeVisible();
  });

  it("should not display user info section and logout button when user session is not existed", () => {
    jest.spyOn(nextAuth, "useSession").mockReturnValue({ data: null, status: "unauthenticated" } as any);

    renderNavbar();

    expect(screen.getByTestId("navbar")).toBeVisible();
    expect(screen.getByTestId("navbar-title")).toHaveTextContent("User Management");
    expect(screen.queryByTestId("user-info")).not.toBeInTheDocument();
    expect(screen.queryByTestId("logout-button")).not.toBeInTheDocument();
  });
});
