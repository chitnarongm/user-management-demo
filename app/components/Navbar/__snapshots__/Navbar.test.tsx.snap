// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-m99cae"
    data-testid="navbar"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-1f8a19r-MuiTypography-root"
          data-testid="navbar-title"
        >
          User Management
        </div>
        <div
          class="MuiBox-root css-70qvj9"
        >
          <div
            class="MuiBox-root css-3cyd2n"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1pnv3kc-MuiTypography-root"
              data-testid="user-info"
            >
              mockuser@email.com
            </h6>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
            data-testid="logout-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LogoutIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4z"
                />
              </svg>
            </span>
            Log Out
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
